lib/autoproj/cli/main.rb

```diff
@@ -343,12 +343,12 @@ def tag(tag_name = nil, *packages)
                 run_autoproj_cli(:tag, :Tag, Hash[], tag_name, *packages)
             end
 
-            desc 'commit [PACKAGES]', 'save the package current versions as a new commit in the main build configuration'
+            desc 'commit [TAG_NAME] [PACKAGES]', 'save the package current versions as a new commit in the main build configuration'
             long_desc <<-EOD
 The commit subcommand stores the state of all packages (or of the packages
 selected on the command line) into a new commit in the currently checked-out
 branch of the build configuration. This state can be retrieved later on by using
-"autoproj reset"
+"autoproj reset". If a TAG_NAME is provided, the commit will be tagged.
```

Comment:
Sorry, I just realized this.

You cannot provide a package without a tag anymore, since the first argument is always interpreted as a tag.

You would either need to reintroduce the `tag` subcommand, or add a `--tag` option to `commit`

Reply:
The tag subcommand has the same issue. I assumed it was a known limitation. Will fix both

Reply:
Nevermind.. [TAG_NAME] is not really optional for the tag subcommand.

Reply:
Done.

Reply:
Wait ... if there is a `tag` subcommand ... why are you introducing the ability to create a tag in `commit` ? Or do you think it's better this way and that we should deprecate `tag` ? I'm not sure myself, as we still need a command to list the tags ...

Reply:
Maybe merge both.. Currently, `tag` doesn't commit and `commit` doesn't tag :P

Reply:
`tag` just creates a dangling commit...

Reply:
Or, deprecate `tag`. I don't see a good use case for that even though they do different things...

Reply:
Alright, now I see it. Thanks for the refresher between `commit` and `tag`

`tag` does not create a dangling commit, it creates a `tag` WITHOUT changing the current branch. Which is great to tag "known good states" without pinning the branch - i.e. letting everyone continue working normally. 

`commit` publishes a certain state to the branch. In the time where I was still using those, `tag` was actually a lot more useful for me than `commit.

I do agree that having the new `--tag` option to `commit` is useful, so I'm still fine with this PR. But the two commands are still very different.

Reply:
With `tag` the generated version file is still commited just not to any branch (otherwise tagging wouldn't be possible). That what I meant by a dangling commit.

> I do agree that having the new --tag option to commit is useful, so I'm still fine with this PR. But the two commands are still very different.

How would you like to proceed with this?

Reply:
> That what I meant by a dangling commit.

Got that ... But then sorry for being pedantic, but "dangling commit" in Git really means "not attached to a branch or tag". The generated commit for instance won't show up in a `git fsck`.